modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +561,180 @@ class ApiClient(object):
+        b64_signed_msg = self.sign_digest(digest)
+
+        auth_header = self.get_authorization_header(signed_headers, b64_signed_msg, created)
+        signed_header_dict['Authorization'] = '{0}'.format(auth_header)
+
+        return signed_header_dict
+
+    def get_message_digest(self, data):
+        """
+        Calculates and returns a cryptographic digest of a specified HTTP request.
+
+        :param data: The message to be hashed with a cryptographic hash.
+        :return: The message digest encoded as a byte string. 
+        """
+        if self.configuration.signing_scheme in ["rsa-sha512", "hs2019"]:
+            digest = SHA512.new()
+            prefix = "SHA-512="
+        elif self.configuration.signing_scheme in ["rsa-sha256"]:
+            digest = SHA256.new()
+            prefix = "SHA-256="
+        else:
+            raise Exception(
+                "Unsupported signing algorithm: {0}".format(self.configuration.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the configuration.
+
+        :param digest: digest of the HTTP message.
+        :return: the HTTP message signature encoded as a byte string.
+        """
+        self.configuration.load_private_key()
+        privkey = self.configuration.private_key
+        if isinstance(privkey, RSA.RsaKey):
+            if self.configuration.signing_algorithm == 'PSS':
+                # RSASSA-PSS in Section 8.1 of RFC8017.
+                signature = pss.new(privkey).sign(digest)
+            elif self.configuration.signing_algorithm == 'PKCS1v15':
+                # RSASSA-PKCS1-v1_5 in Section 8.2 of RFC8017.
+                signature = PKCS1_v1_5.new(privkey).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        elif isinstance(privkey, ECC.EccKey):
+            if self.configuration.signing_algorithm in ['fips-186-3', 'deterministic-rfc6979']:
+                signature = DSS.new(privkey, self.configuration.signing_algorithm).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        else:
+            raise Exception("Unsupported private key: {0}".format(type(privkey)))
+        return b64encode(signature)
+
+    def get_str_to_sign(self, signed_headers):
+        """
+        Generate and return a string value representing the HTTP request to be signed.
+
+        :param signed_headers: The HTTP Headers to be signed.
+        :return: instance of digest object 
+        """
+        ss = ""
+        i = 0
+        for key, value in signed_headers.items():
+            if i > 0:
+                ss = ss + "\n"
+            ss = ss + key.lower() + ": " + value
```

Comment:
Similar comment to what I say below:
Can you move this line higher than the if i > 0?
It is confusing because this lower line is run in the i==0 use case.
What about using this?
```
            ss += "{0}:{1}".format(key.lower(), value)
            if i > 0:
                ss += "\n"
```

Reply:
doing.